contracts/2key/campaign-mutual-contracts/TwoKeyCampaign.sol

```diff
@@ -309,31 +315,39 @@ contract TwoKeyCampaign is ArcERC20 {
  	 * @dev It can be called by the address specified in the param or by the one of two key maintainers
  	 */
 	function referrerWithdraw(
-		address _address
+		address _address,
+		bool _withdrawAsStable
 	)
 	public
 	{
 		require(msg.sender == _address || twoKeyEventSource.isAddressMaintainer(msg.sender));
 		address twoKeyAdminAddress;
 		address twoKeyUpgradableExchangeContract;
 		uint balance;
+		uint twoKeyRealeaseDate;
 
 		address _referrer = twoKeyEventSource.plasmaOf(_address);
 		if(referrerPlasma2Balances2key[_referrer] != 0) {
-			twoKeyAdminAddress =  getContractProxyAddress("TwoKeyAdmin");
+			twoKeyAdminAddress = getContractProxyAddress("TwoKeyAdmin");
 			twoKeyUpgradableExchangeContract = getContractProxyAddress("TwoKeyUpgradableExchange");
 
 			balance = referrerPlasma2Balances2key[_referrer];
 			referrerPlasma2Balances2key[_referrer] = 0;
 
-			if(now >= ITwoKeyAdmin(twoKeyAdminAddress).getTwoKeyRewardsReleaseDate()){
-				IERC20(twoKeyEconomy).transfer(_address,balance);
-			}
-			else {
-				//In case 2Key rewards still locked;
+			if(_withdrawAsStable == true){
 				IERC20(twoKeyEconomy).approve(twoKeyUpgradableExchangeContract, balance);
 				IUpgradableExchange(twoKeyUpgradableExchangeContract).buyStableCoinWith2key(balance, _address);
 			}
+			else{
+				twoKeyRealeaseDate = ITwoKeyAdmin(twoKeyAdminAddress).getTwoKeyRewardsReleaseDate();
+				if (block.timestamp < twoKeyRealeaseDate){
+					emit TokensLocked(twoKeyRealeaseDate, balance);
+					revert();
```

Comment:
In case we're emitting an event which will let us know what's going on, no need to *revert()* function, line 345, enough to return (break) execution and we can understand from parsed event what happened and show that message to user

Reply:
I think it will cause problems because right before that function we are resetting user's reward balance to ZERO so if we'll quit without revert we changed the state without sending the rewards to the converter. 

This are the relevant lines (334-335)
balance = referrerPlasma2Balances2key[_referrer];
referrerPlasma2Balances2key[_referrer] = 0;